content-src/lib/selectLayoutRender.js

```diff
@@ -94,4 +106,24 @@ export const selectLayoutRender = (state, prefs, rickRollCache) => {
       return {...component, data};
     }),
   })).filter(row => row.components.length);
+
+  let spocsFill = [];
+  if (spocs.data.spocs) {
```

Comment:
Is this the proper if statement?

I wonder if we want to instead check if spocs are enabled, and if the has already been sent? Because I *think* if there are no spocs, this could be because the server is out of spocs or maybe we had no available spocs left?

Reply:
I wonder if this has something to do with the `fullRecalc` stuff you have in the feed.jsm?

Reply:
That's correct, for fullRecalc case we need to ensure spoc data is loaded on first fetch or a layout reset, so we either keep this check or even better check for spocs.loaded similar to https://github.com/mozilla/activity-stream/blob/master/common/Reducers.jsm#L464

Reply:
Hmm, this `if` has nothing to do with `fullRecalc`, rather it is for the `spocs.data.spocs.slice(spocIndex)` down below. Also, if spocs is not enabled, shouldn't `spocs.data` be an empty object `{}`? Correct me if I was wrong here, I might've made a wrong assumption.

Reply:
That's correct, looking at loadSpocs in DiscoveryStreamFeed.jsm, spocs.data is loaded from cache or endpoint only if spocs are enabled else data is empty